chains/evm/contracts/offRamp/CCVAggregator.sol

```diff
@@ -208,10 +209,127 @@ contract CCVAggregator is ITypeAndVersion, Ownable2StepMsgSender {
+  //
+  //    if (success) {
+  //      // If message execution succeeded, no CCIP receiver return data is expected, return with empty bytes.
+  //      return (Internal.MessageExecutionState.SUCCESS, "");
+  //    }
+  //
+  //    // We want to detect if we ran out of gas inside the CallWithGasBuffer to allow for better error handling.
+  //    // EIP-150 means only 63/64 of the gas is forwarded to the callee. If they ran out of gas, the 1/64 is available
+  //    // in this function again and should not be counted towards the gas that was left inside the function.
+  //    // 20_000 gas is an additional buffer as the contract ran out of gas before it could do some action that would have
+  //    // taken it over the gas allowance. The most expensive operation that is regularly done is an SSTORE which costs
+  //    // 20_000 gas when setting a storage slot from zero to non-zero. This quite large buffer does mean there could be
+  //    // false positives.
+  //    bool ranOutOfGasInCall = err.length == 0 && gasleft() < minimumGasLeft + 20_000;
+  //
+  //    // We have a special gas estimator caller address that is used to cause reverts on Out Of Gas situations. Normally,
+  //    // the call would succeed but mark the message as failed. However, for gas estimation we want to know if the
+  //    // message would have succeeded or not, so we revert here to signal to the gas estimator that the gas limit was
+  //    // insufficient.
+  //    if (msg.sender == Internal.GAS_ESTIMATION_SENDER) {
+  //      if (
+  //        CallWithExactGas.NOT_ENOUGH_GAS_FOR_CALL_SIG == bytes4(err)
+  //          || CallWithExactGas.NO_GAS_FOR_CALL_EXACT_CHECK_SIG == bytes4(err)
+  //          || ERC165CheckerReverting.InsufficientGasForStaticCall.selector == bytes4(err) || ranOutOfGasInCall
+  //      ) {
+  //        revert InsufficientGasToCompleteTx(bytes4(err));
+  //      }
+  //    }
+  //    // We don't compare against gasBufferToUpdateState as we know there was enough gas to perform the call, so
+  //    // we expect it to be spent. This reduces false positives in the out of gas detection.
+  //    if (ranOutOfGasInCall) {
+  //      err = abi.encodeWithSelector(OutOfGas.selector);
+  //    }
+  //
+  //    // return the message execution state as FAILURE and the revert data.
+  //    // Max length of revert data is Router.MAX_RET_BYTES, max length of err is 4 + Router.MAX_RET_BYTES.
+  //    return (Internal.MessageExecutionState.FAILURE, err);
+  //  }
+
+  function _callWithGasBuffer(
+    bytes memory payload
+  ) internal returns (bool success, bytes memory retData, uint256 minimumGasLeft) {
+    // allocate retData memory ahead of time
+    retData = new bytes(Internal.MAX_RET_BYTES);
+    uint16 maxReturnBytes = Internal.MAX_RET_BYTES;
+
+    uint256 gasLeft = gasleft();
+    // We add 100 gas as a buffer for the operations after the gasleft() call on the line above.
```

Comment:
So MAX_GAS_BUFFER_TO_UPDATE_STATE accounts for the state update required after _callWithGasBuffer returns, and the 100 accounts for the gas required for the remainder of this function?